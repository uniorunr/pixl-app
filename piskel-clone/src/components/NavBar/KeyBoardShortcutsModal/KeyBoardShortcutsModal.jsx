import React, { Component } from 'react';
import PropTypes from 'prop-types';
import ShortcutItem from './ShortcutItem/ShortcutItem';
import './KeyBoardShortcutsModal.scss';

class ShortcutsModal extends Component {
  state = {
    tools: [],
    activeTool: 0,
  };

  makeActive = (index) => {
    const { updateActiveTool } = this.props;
    const { tools } = this.state;
    this.setState({
      activeTool: index,
    });
    updateActiveTool(tools[index].dataset.tool);
  };

  render() {
    const { toggleModal, toolsData } = this.props;
    const { tools, activeTool } = this.state;

    return (
      <section className="shortcuts-modal">
        <div className="shortcuts-modal__content">
          <h2>Keyboard shortcuts</h2>
          <p>Click on tool and press the key to change the shortcut</p>
          <div className="shortcuts-modal__shortcut-rows">
            {Object.keys(toolsData).map((id, index) => (
              <ShortcutItem
                iconClass={toolsData[id].iconClass}
                toolName={toolsData[id].name}
                key={id}
                active={index === activeTool}
                tools={tools}
                index={index}
                makeActive={this.makeActive}
                keyCode={toolsData[id].shortcut}
                id={id}
              />
            ))}
          </div>
        </div>
        <button
          type="button"
          className="shortcuts-modal__close"
          onClick={toggleModal}
        >
          <span>x</span>
        </button>
      </section>
    );
  }
}

ShortcutsModal.propTypes = {
  toggleModal: PropTypes.func.isRequired,
  toolsData: PropTypes.instanceOf(Object).isRequired,
  updateActiveTool: PropTypes.func.isRequired,
};

export default ShortcutsModal;
