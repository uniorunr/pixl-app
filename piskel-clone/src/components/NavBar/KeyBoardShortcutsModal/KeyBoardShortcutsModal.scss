.shortcuts-modal {
  border: 2px solid black;
  margin: auto;
  left: 0;
  right: 0;
  top: 0;
  bottom: 0;
  width: 95%;
  height: 95%;
  position: absolute;
  background: white;
  z-index: 4;
  display: flex;
  justify-content: center;
  align-items: center;

  .shortcuts-modal__content{
    width: 90%;
    height: 90%;

    p {
      text-align: center;
      font-size: 1.4rem;
    }

    h2 {
      text-align: center;
      font-size: 1.6rem;
      margin: 0;
    }

    .shortcuts-modal__shortcut-rows {
      display: grid;
      grid-template-columns: 1fr 1fr;
      grid-column-gap: 100px;
      grid-row-gap: 20px;
      margin: 20px 100px 0 150px;
      padding: 15px;
      font-weight: 700;
      font-size: 1.4rem;
      background-color: white;
    }
  }

  .shortcuts-modal__close {
    position: absolute;
    top: -2px;
    right: -2px;
    border: 2px solid black;
    outline: none;
    background: white;
    cursor: pointer;

    span {
      font-weight: 700;
      font-size: 1.6rem;
    }
  }

  .shortcuts-modal__close:hover {
    border: 2px dashed black;
  }
}
