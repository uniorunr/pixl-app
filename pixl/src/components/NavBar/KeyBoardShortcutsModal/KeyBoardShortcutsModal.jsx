import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import * as actions from '../../../actions/actions';
import ToolShortcutItem from './ToolShortcutItem/ToolShortcutItem';
import FrameShortcutItem from './FrameShortcutItem/FrameShortcutItem';
import LayerShortcutItem from './LayerShortcutItem/LayerShortcutItem';
import './KeyBoardShortcutsModal.scss';

class ShortcutsModal extends Component {
  makeActive = (index, type) => {
    const {
      updateActiveTool,
      updateActiveToolIndex,
      updateActiveFrameShortcut,
      updateActiveFrameShortcutIndex,
      updateActiveLayerShortcut,
      updateActiveLayerShortcutIndex,
      activeBlock,
      tools,
      frames,
      layers,
    } = this.props;

    if (activeBlock === 'tools' && type === 'tools') {
      updateActiveTool(tools[index].dataset.tool);
      updateActiveToolIndex(index);
    } else if (activeBlock === 'frames' && type === 'frames') {
      updateActiveFrameShortcut(frames[index].dataset.action);
      updateActiveFrameShortcutIndex(index);
    } else if (activeBlock === 'layers' && type === 'layers') {
      updateActiveLayerShortcutIndex(index);
      updateActiveLayerShortcut(layers[index].dataset.action);
    }
  };

  clickOnBlock = ({ target }) => {
    const { updateActiveBlock } = this.props;
    let currTarget = target;
    while (!currTarget.dataset.block) {
      currTarget = currTarget.parentElement;
    }
    updateActiveBlock(currTarget.dataset.block);
  };

  render() {
    const {
      updateShortcutsModalState,
      toolsData,
      framesShortcuts,
      activeBlock,
      layersShortcuts,
      activeToolIndex,
      activeFrameShortcutIndex,
      activeLayerShortcutIndex,
    } = this.props;

    return (
      <section className="shortcuts-modal">
        <div className="shortcuts-modal__content">
          <h2>Keyboard shortcuts</h2>
          <p>Click on tool and press the key to change the shortcut</p>
          <h3>Tools</h3>
          <div
            role="presentation"
            className={`shortcuts-modal__shortcut-rows ${
              activeBlock === 'tools'
                ? 'shortcuts-modal__shortcut-rows_active'
                : ''
            }`}
            data-block="tools"
            onClick={this.clickOnBlock}
          >
            {Object.keys(toolsData).map((id, index) => (
              <ToolShortcutItem
                key={id}
                active={index === activeToolIndex}
                index={index}
                makeActive={this.makeActive}
                id={id}
              />
            ))}
          </div>
          <h3>Frames</h3>
          <div
            role="presentation"
            className={`shortcuts-modal__shortcut-rows ${
              activeBlock === 'frames'
                ? 'shortcuts-modal__shortcut-rows_active'
                : ''
            }`}
            onClick={this.clickOnBlock}
            data-block="frames"
          >
            {Object.keys(framesShortcuts).map((id, index) => (
              <FrameShortcutItem
                active={index === activeFrameShortcutIndex}
                id={id}
                key={id}
                index={index}
                makeActive={this.makeActive}
              />
            ))}
          </div>
          <h3>Layers</h3>
          <div
            role="presentation"
            className={`shortcuts-modal__shortcut-rows ${
              activeBlock === 'layers'
                ? 'shortcuts-modal__shortcut-rows_active'
                : ''
            }`}
            onClick={this.clickOnBlock}
            data-block="layers"
          >
            {Object.keys(layersShortcuts).map((id, index) => (
              <LayerShortcutItem
                active={index === activeLayerShortcutIndex}
                id={id}
                key={id}
                index={index}
                makeActive={this.makeActive}
              />
            ))}
          </div>
        </div>
        <button
          type="button"
          className="shortcuts-modal__close"
          onClick={updateShortcutsModalState}
        >
          <span>x</span>
        </button>
      </section>
    );
  }
}

ShortcutsModal.propTypes = {
  updateShortcutsModalState: PropTypes.func.isRequired,
  toolsData: PropTypes.instanceOf(Object).isRequired,
  updateActiveTool: PropTypes.func.isRequired,
  updateActiveToolIndex: PropTypes.func.isRequired,
  updateActiveBlock: PropTypes.func.isRequired,
  updateActiveFrameShortcut: PropTypes.func.isRequired,
  updateActiveFrameShortcutIndex: PropTypes.func.isRequired,
  updateActiveLayerShortcut: PropTypes.func.isRequired,
  updateActiveLayerShortcutIndex: PropTypes.func.isRequired,
  activeBlock: PropTypes.string.isRequired,
  framesShortcuts: PropTypes.instanceOf(Object).isRequired,
  layersShortcuts: PropTypes.instanceOf(Object).isRequired,
  activeToolIndex: PropTypes.number.isRequired,
  activeFrameShortcutIndex: PropTypes.number.isRequired,
  activeLayerShortcutIndex: PropTypes.number.isRequired,
  tools: PropTypes.instanceOf(Array).isRequired,
  frames: PropTypes.instanceOf(Array).isRequired,
  layers: PropTypes.instanceOf(Array).isRequired,
};

const mapStateToProps = state => ({
  toolsData: state.tools.toolsData,
  framesShortcuts: state.frames.framesShortcuts,
  layersShortcuts: state.layers.layersShortcuts,
  activeBlock: state.components.navBar.activeBlock,
  activeToolIndex: state.components.navBar.modalWindow.activeToolIndex,
  activeFrameShortcutIndex:
    state.components.navBar.modalWindow.activeFrameShortcutIndex,
  activeLayerShortcutIndex:
    state.components.navBar.modalWindow.activeLayerShortcutIndex,
  tools: state.components.navBar.modalWindow.toolsShortcutsRefs,
  frames: state.components.navBar.modalWindow.framesShortcutsRefs,
  layers: state.components.navBar.modalWindow.layersShortcutsRefs,
});

const mapDispatchToProps = (dispatch) => {
  const {
    updateShortcutsModalState,
    updateActiveBlock,
    updateActiveTool,
    updateActiveToolIndex,
    updateActiveFrameShortcut,
    updateActiveFrameShortcutIndex,
    updateActiveLayerShortcut,
    updateActiveLayerShortcutIndex,
  } = bindActionCreators(actions, dispatch);
  return {
    updateShortcutsModalState,
    updateActiveBlock: (block) => {
      updateActiveBlock(block);
    },
    updateActiveTool: (id) => {
      updateActiveTool(id);
    },
    updateActiveToolIndex: (index) => {
      updateActiveToolIndex(index);
    },
    updateActiveFrameShortcut: (id) => {
      updateActiveFrameShortcut(id);
    },
    updateActiveFrameShortcutIndex: (index) => {
      updateActiveFrameShortcutIndex(index);
    },
    updateActiveLayerShortcut: (id) => {
      updateActiveLayerShortcut(id);
    },
    updateActiveLayerShortcutIndex: (index) => {
      updateActiveLayerShortcutIndex(index);
    },
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(ShortcutsModal);
