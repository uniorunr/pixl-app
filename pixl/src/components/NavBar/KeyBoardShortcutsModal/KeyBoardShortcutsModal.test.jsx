import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import KeyBoardShortcutsModal from './KeyBoardShortcutsModal';
import store from '../../../stores/mainStore';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <KeyBoardShortcutsModal />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
