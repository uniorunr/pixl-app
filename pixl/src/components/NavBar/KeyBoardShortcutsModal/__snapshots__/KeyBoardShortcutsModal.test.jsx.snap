// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  className="shortcuts-modal"
>
  <div
    className="shortcuts-modal__content"
  >
    <h2>
      Keyboard shortcuts
    </h2>
    <p>
      Click on tool and press the key to change the shortcut
    </p>
    <h3>
      Tools
    </h3>
    <div
      className="shortcuts-modal__shortcut-rows shortcuts-modal__shortcut-rows_active"
      data-block="tools"
      onClick={[Function]}
      role="presentation"
    >
      <div
        className="shortcuts-modal__row"
        onClick={[Function]}
        role="presentation"
      >
        <div
          className="shortcuts-modal__tool shortcuts-modal__tool_active"
          data-tool="pen"
        >
          <i
            className="fas fa-pen"
          />
          <span>
            Pen tool
          </span>
          <span>
            -
          </span>
          <span
            className="shortcuts-modal__shortcut"
          >
            KeyP
          </span>
        </div>
      </div>
      <div
        className="shortcuts-modal__row"
        onClick={[Function]}
        role="presentation"
      >
        <div
          className="shortcuts-modal__tool "
          data-tool="vertical-mirror"
        >
          <i
            className="fas fa-grip-lines-vertical"
          />
          <span>
            Vertical mirror tool
          </span>
          <span>
            -
          </span>
          <span
            className="shortcuts-modal__shortcut"
          >
            KeyV
          </span>
        </div>
      </div>
      <div
        className="shortcuts-modal__row"
        onClick={[Function]}
        role="presentation"
      >
        <div
          className="shortcuts-modal__tool "
          data-tool="horizontal-mirror"
        >
          <i
            className="fas fa-grip-lines"
          />
          <span>
            Horizontal Mirror tool
          </span>
          <span>
            -
          </span>
          <span
            className="shortcuts-modal__shortcut"
          >
            KeyH
          </span>
        </div>
      </div>
      <div
        className="shortcuts-modal__row"
        onClick={[Function]}
        role="presentation"
      >
        <div
          className="shortcuts-modal__tool "
          data-tool="stroke"
        >
          <i
            className="fas fa-slash"
          />
          <span>
            Stroke tool
          </span>
          <span>
            -
          </span>
          <span
            className="shortcuts-modal__shortcut"
          >
            KeyS
          </span>
        </div>
      </div>
      <div
        className="shortcuts-modal__row"
        onClick={[Function]}
        role="presentation"
      >
        <div
          className="shortcuts-modal__tool "
          data-tool="rectangle"
        >
          <i
            className="far fa-square"
          />
          <span>
            Rectangle tool
          </span>
          <span>
            -
          </span>
          <span
            className="shortcuts-modal__shortcut"
          >
            KeyR
          </span>
        </div>
      </div>
      <div
        className="shortcuts-modal__row"
        onClick={[Function]}
        role="presentation"
      >
        <div
          className="shortcuts-modal__tool "
          data-tool="circle"
        >
          <i
            className="far fa-circle"
          />
          <span>
            Circle tool
          </span>
          <span>
            -
          </span>
          <span
            className="shortcuts-modal__shortcut"
          >
            KeyC
          </span>
        </div>
      </div>
      <div
        className="shortcuts-modal__row"
        onClick={[Function]}
        role="presentation"
      >
        <div
          className="shortcuts-modal__tool "
          data-tool="paint-bucket"
        >
          <i
            className="fas fa-fill-drip"
          />
          <span>
            Paint bucket tool
          </span>
          <span>
            -
          </span>
          <span
            className="shortcuts-modal__shortcut"
          >
            KeyB
          </span>
        </div>
      </div>
      <div
        className="shortcuts-modal__row"
        onClick={[Function]}
        role="presentation"
      >
        <div
          className="shortcuts-modal__tool "
          data-tool="paint-same-pixels"
        >
          <i
            className="fas fa-paint-roller"
          />
          <span>
            Paint same pixels tool
          </span>
          <span>
            -
          </span>
          <span
            className="shortcuts-modal__shortcut"
          >
            KeyA
          </span>
        </div>
      </div>
      <div
        className="shortcuts-modal__row"
        onClick={[Function]}
        role="presentation"
      >
        <div
          className="shortcuts-modal__tool "
          data-tool="choose-color"
        >
          <i
            className="fas fa-eye-dropper"
          />
          <span>
            Color picker tool
          </span>
          <span>
            -
          </span>
          <span
            className="shortcuts-modal__shortcut"
          >
            KeyY
          </span>
        </div>
      </div>
      <div
        className="shortcuts-modal__row"
        onClick={[Function]}
        role="presentation"
      >
        <div
          className="shortcuts-modal__tool "
          data-tool="eraser"
        >
          <i
            className="fas fa-eraser"
          />
          <span>
            Eraser tool
          </span>
          <span>
            -
          </span>
          <span
            className="shortcuts-modal__shortcut"
          >
            KeyE
          </span>
        </div>
      </div>
      <div
        className="shortcuts-modal__row"
        onClick={[Function]}
        role="presentation"
      >
        <div
          className="shortcuts-modal__tool "
          data-tool="move"
        >
          <i
            className="fas fa-arrows-alt"
          />
          <span>
            Move tool
          </span>
          <span>
            -
          </span>
          <span
            className="shortcuts-modal__shortcut"
          >
            KeyM
          </span>
        </div>
      </div>
      <div
        className="shortcuts-modal__row"
        onClick={[Function]}
        role="presentation"
      >
        <div
          className="shortcuts-modal__tool "
          data-tool="lighten"
        >
          <i
            className="fas fa-sun"
          />
          <span>
            Lighten tool
          </span>
          <span>
            -
          </span>
          <span
            className="shortcuts-modal__shortcut"
          >
            KeyL
          </span>
        </div>
      </div>
    </div>
    <h3>
      Frames
    </h3>
    <div
      className="shortcuts-modal__shortcut-rows "
      data-block="frames"
      onClick={[Function]}
      role="presentation"
    >
      <div
        className="shortcuts-modal__row"
        data-action="duplicate"
        onClick={[Function]}
        role="presentation"
      >
        <div
          className="shortcuts-modal__tool shortcuts-modal__tool_active"
          data-action="duplicate"
        >
          <i
            className="fas fa-copy"
          />
          <span>
            Duplicate Frame
          </span>
          <span>
            -
          </span>
          <span
            className="shortcuts-modal__shortcut"
          >
            <span
              className="shortcuts-modal__prefix"
            >
              shiftKey + 
            </span>
            KeyF
          </span>
        </div>
      </div>
      <div
        className="shortcuts-modal__row"
        data-action="remove"
        onClick={[Function]}
        role="presentation"
      >
        <div
          className="shortcuts-modal__tool "
          data-action="remove"
        >
          <i
            className="fas fa-trash-alt"
          />
          <span>
            Remove Frame
          </span>
          <span>
            -
          </span>
          <span
            className="shortcuts-modal__shortcut"
          >
            <span
              className="shortcuts-modal__prefix"
            >
              shiftKey + 
            </span>
            KeyR
          </span>
        </div>
      </div>
    </div>
    <h3>
      Layers
    </h3>
    <div
      className="shortcuts-modal__shortcut-rows "
      data-block="layers"
      onClick={[Function]}
      role="presentation"
    >
      <div
        className="shortcuts-modal__row"
        data-action="add"
        onClick={[Function]}
        role="presentation"
      >
        <div
          className="shortcuts-modal__tool shortcuts-modal__tool_active"
          data-action="add"
        >
          <i
            className="fas fa-plus"
          />
          <span>
            Add layer
          </span>
          <span>
            -
          </span>
          <span
            className="shortcuts-modal__shortcut"
          >
            <span
              className="shortcuts-modal__prefix"
            >
              altKey + 
            </span>
            KeyL
          </span>
        </div>
      </div>
      <div
        className="shortcuts-modal__row"
        data-action="moveUp"
        onClick={[Function]}
        role="presentation"
      >
        <div
          className="shortcuts-modal__tool "
          data-action="moveUp"
        >
          <i
            className="fas fa-arrow-up"
          />
          <span>
            Move Up
          </span>
          <span>
            -
          </span>
          <span
            className="shortcuts-modal__shortcut"
          >
            <span
              className="shortcuts-modal__prefix"
            >
              altKey + 
            </span>
            ArrowUp
          </span>
        </div>
      </div>
      <div
        className="shortcuts-modal__row"
        data-action="moveDown"
        onClick={[Function]}
        role="presentation"
      >
        <div
          className="shortcuts-modal__tool "
          data-action="moveDown"
        >
          <i
            className="fas fa-arrow-down"
          />
          <span>
            Move Down
          </span>
          <span>
            -
          </span>
          <span
            className="shortcuts-modal__shortcut"
          >
            <span
              className="shortcuts-modal__prefix"
            >
              altKey + 
            </span>
            ArrowDown
          </span>
        </div>
      </div>
      <div
        className="shortcuts-modal__row"
        data-action="remove"
        onClick={[Function]}
        role="presentation"
      >
        <div
          className="shortcuts-modal__tool "
          data-action="remove"
        >
          <i
            className="fas fa-trash-alt"
          />
          <span>
            Remove Layer
          </span>
          <span>
            -
          </span>
          <span
            className="shortcuts-modal__shortcut"
          >
            <span
              className="shortcuts-modal__prefix"
            >
              altKey + 
            </span>
            KeyR
          </span>
        </div>
      </div>
      <div
        className="shortcuts-modal__row"
        data-action="merge"
        onClick={[Function]}
        role="presentation"
      >
        <div
          className="shortcuts-modal__tool "
          data-action="merge"
        >
          <i
            className=""
          />
          <span>
            Merge Layers
          </span>
          <span>
            -
          </span>
          <span
            className="shortcuts-modal__shortcut"
          >
            <span
              className="shortcuts-modal__prefix"
            >
              altKey + 
            </span>
            KeyM
          </span>
        </div>
      </div>
    </div>
  </div>
  <button
    className="shortcuts-modal__close"
    onClick={[Function]}
    type="button"
  >
    <span>
      x
    </span>
  </button>
</section>
`;
